openedx/course-discovery
# fix: geolocation serializer fixups

### [PROD-3048](https://2u-internal.atlassian.net/browse/PROD-3048)

### Description
- Allow GeoLocation fields to be null. This is to accommodate GeoLocation data coming from Publisher. GeoLoc info on Publisher is optional. Related PR: https://github.com/openedx/frontend-app-publisher/pull/819
- Update courses serializer to skip geolocation handling if all fields are not present. All of the geoloc fields should be provided to keep the data consistent. 
- Fix the update geoloc behavior where it would always create a new geoloc object if any of the fields is edited on the publisher. This results in having a lot of orphan GeoLoc objects in DB. 
- UUID filtering flaky test fixup



### Testing
You would need publisher running for proper testing.
- Checkout this PR and publisher's PR https://github.com/openedx/frontend-app-publisher/pull/819 on your local
- Go to any course on the publisher. You should see Geolocation section on Course Edit section
- Do not add any values in any of the field. Update any other field of Course/Course run so that backend call is made. Ensure no geoLoc object is created
- Add value for all 3 geoloc fields and save/publish again. Ensure geoloc is created. If the course is already published, ensure that geoloc object is linked with both draft and non-draft versions of the course.
- Update any of geoloc fields and save again. Ensure that no new GeoLoc object is created and that existing GeoLoc object related to the course is updated.
